src/pages/en/reference/integrations-reference.md

```diff
@@ -40,30 +39,159 @@ interface AstroIntegration {
+
+#### "command" option
+
+**Type:** `'dev' / 'build'`
+
+- `dev` - project was executed with `astro dev` or `astro preview`
+- `build` - project was executed with `astro build`
+
+#### "updateConfig" option
+
+**Type:** `(newConfig: Record<string, any>) => void;`
+
+A callback function to update the user-supplied [Astro config](/en/reference/configuration-reference/). Any config you provide **will be merged with the user config,** so you are free to omit keys! 
+
+For example, say you need to supply a [Vite](https://vitejs.dev) plugin to the user's project:
+
+```js
+import bananaCSS from '@vitejs/official-banana-css-plugin';
+
+export default {
+  name: 'banana-css-integration',
+  hooks: {
+    'astro:config:setup': ({ updateConfig }) => {
+      updateConfig({
+        vite: {
+          plugins: [bananaCSS()],
+        }
+      })
+    }
+  }
+}
+```
+
+#### "addRenderer" option
+
+- **Type:** [`AstroRenderer`](https://github.com/withastro/astro/blob/fdd607c5755034edf262e7b275732519328a33b2/packages/astro/src/%40types/astro.ts#L872-L883)
+- **Examples:** [`lit`](https://github.com/withastro/astro/blob/main/packages/integrations/lit/src/index.ts), [`svelte`](https://github.com/withastro/astro/blob/main/packages/integrations/svelte/src/index.ts), [`react`](https://github.com/withastro/astro/blob/main/packages/integrations/react/src/index.ts), [`preact`](https://github.com/withastro/astro/blob/main/packages/integrations/preact/src/index.ts), [`vue`](https://github.com/withastro/astro/blob/main/packages/integrations/vue/src/index.ts), [`solid`](https://github.com/withastro/astro/blob/main/packages/integrations/solid/src/index.ts)
+
+A callback function to add a component framework renderer (i.e. React, Vue, Svelte, etc). You can browse the examples and type definition above for more advanced options, but here are the 2 main options to be aware of:
+- `clientEntrypoint` - path to a file that executes on the client whenever your component is used. This is mainly for rendering or hydrating your component with JS.
+- `serverEntrypoint` - path to a file that executes during server-side requests or static builds whenever your component is used. These should render components to static markup, with hooks for hydration where applicable. [React's `renderToString` callback](https://reactjs.org/docs/react-dom-server.html#rendertostring) is a classic example.
+
+#### "injectScript" option
+
+**Type:** `(stage: InjectedScriptStage, content: string) => void;`
+
+A callback function to inject a string of JavaScript content onto every page.
+
+The **`stage`** denotes how this script (the `content`) should be inserted. Some stages are inserted purely, while others are picked up by Vite:
```

Comment:
Overall, this doc update is really fantastic!

I seem to get a bit hung up on some of the Vite specific stuff and I could be wrong but there's some assumed background knowledge there. I don't think you need to increase the scope to this doc, but linking to Vite resources to help get any prereq knowledge may be helpful.

> while others are picked up by Vite

What's the significance of this. Probably dumb question for those w/more understanding of Vite, but this is an example from this noob ;-)

Reply:
Totally understand! I updated this to say:

> The **`stage`** denotes how this script (the `content`) should be inserted. Some stages allow inserting scripts without modification, while others allow optimization during [Vite's bundling step](https://vitejs.dev/guide/build.html)

Let me know if this helps!